packit_service/worker/handlers/distgit.py

```diff
@@ -284,31 +300,63 @@ def run(self) -> TaskResults:
                     ProposeDownstreamTargetStatus.queued,
                 ]:
                     continue
-
+                logger.debug(f"Running propose downstream for {branch}")
                 model.set_status(status=ProposeDownstreamTargetStatus.running)
+                url = get_propose_downstream_info_url(model.id)
                 buffer, handler = gather_packit_logs_to_buffer(
                     logging_level=logging.DEBUG
                 )
 
                 try:
                     model.set_start_time(start_time=datetime.utcnow())
+                    self.propose_downstream_helper.report_status_to_branch(
+                        branch=branch,
+                        description="Starting propose downstream...",
+                        state=BaseCommitStatus.pending,
+                        url=url,
+                    )
                     downstream_pr = self.sync_branch(
                         branch=branch, model=propose_downstream_model
                     )
                     if downstream_pr is not None:
+                        logger.debug("Downstream PR created successfully.")
                         model.set_downstream_pr_url(downstream_pr_url=downstream_pr.url)
                         model.set_status(status=ProposeDownstreamTargetStatus.submitted)
+                        self.propose_downstream_helper.report_status_to_branch(
+                            branch=branch,
+                            description="Propose downstream finished successfully.",
+                            state=BaseCommitStatus.success,
+                            url=url,
+                        )
                 except AbortProposeDownstream:
+                    logger.debug(
+                        "Propose downstream is being retried because "
+                        "we were not able yet to download the archive. "
+                    )
                     model.set_status(staus=ProposeDownstreamTargetStatus.retry)
+                    self.propose_downstream_helper.report_status_to_branch(
+                        branch=branch,
+                        description="Propose downstream is being retried because "
+                        "we were not able yet to download the archive. ",
+                        state=BaseCommitStatus.pending,
+                        url=url,
+                    )
```

Comment:
I like this: was thinking if we should do pending -> fail -> pending -> ...

but that would be confusing: pending -> pending with an accurate message makes more sense